Realm/RLMUtil.mm

```diff
@@ -228,10 +228,24 @@ BOOL RLMIsObjectValidForProperty(__unsafe_unretained id const obj,
 }
 
 NSError *RLMMakeError(RLMError code, const realm::RealmFileException& exception) {
-    NSString *underlying = @(exception.underlying().c_str());
+    // Errors for `open()` now include the path in the exception message, but
+    // RealmFileException's message already has the path, so remove the second
+    // copy as it makes errors much less readable.
+    NSString *underlying;
+    auto pathPos = exception.underlying().find(exception.path());
+    if (pathPos != std::string::npos) {
+        auto trimmed = exception.underlying();
+        trimmed.replace(pathPos, exception.path().size(), "");
+        underlying = @(trimmed.c_str());
+    }
+    else {
+        underlying = @(exception.underlying().c_str());
+    }
+
+    NSString *msg = underlying.length ? [NSString stringWithFormat:@"%s: %@", exception.what(), underlying] : @(exception.what());
```

Comment:
There's a whole bunch of errors that now result in this exception rather than runtime_error, which resulted in the actual error message moving from what() to underlying().